Lib/configparser.py

```diff
@@ -750,6 +751,7 @@ def read(self, filenames, encoding=None):
             if isinstance(filename, os.PathLike):
                 filename = os.fspath(filename)
             read_ok.append(filename)
+        self._loaded_files.extend(read_ok)
```

Comment:
I know it was me who suggested this change. Now that I look more closely, I see that ConfigParser gets its state from more than just read files; there's also `read_file`, `read_string`, and `read_dict`. I wonder if instead, this hook should go in `_read` and add `source` to a list of `_loaded_sources`, and render those, so that one could tell the difference between a ConfigParser that was loaded with a `<string>` or a `source="database-config"` compared to one that hadn't had anything read in.

Reply:
Sure I have made these changes, But what about users directly using set and add_section? I haven’t added anything for this as it's the application code, I think we could skip it.

Reply:
Yes, I'm okay to skip it. The objective is to communicate some details about how the state of the object. I think between the loaded files and the sections summary, that covers most use cases to give a pretty good best effort context.